pkg/virt-controller/services/template.go

```diff
@@ -1052,6 +810,31 @@ func (t *templateService) newVolumeRenderer(vmi *v1.VirtualMachineInstance, name
 	return volumeRenderer, nil
 }
 
+func (t *templateService) newResourceRenderer(vmi *v1.VirtualMachineInstance) (*ResourceRenderer, error) {
+	vmiResources := vmi.Spec.Domain.Resources
+	baseOptions := []ResourceRendererOption{
+		WithEphemeralStorageRequest(),
+		WithVirtualizationResources(getRequiredResources(vmi, t.clusterConfig.AllowEmulation())),
+	}
+
+	networkToResourceMap, err := getNetworkToResourceMap(t.virtClient, vmi)
+	if err != nil {
+		return nil, err
+	}
+
+	if err = validatePermittedHostDevices(&vmi.Spec, t.clusterConfig); err != nil {
+		return nil, err
+	}
+
+	return NewResourceRenderer(
+		vmiResources.Limits,
+		vmiResources.Requests,
+		append(
+			baseOptions,
+			NewVMIPredicates(vmi, t.clusterConfig, networkToResourceMap).Apply()...)...,
```

Comment:
I guess so. 

If we eventually want to have "user provided" predicates, we can move that logic out somehow.

Reply:
On second thought, I really like the fact the `ResourceRenderer` only understands options.

It would be up to the templating service to use something to compute those options from the cluster configuration / vmi spec / etc. That something is (on my proposal) the predicate to resource mapping I've added. 

If I embed the predicate / resource mapping into the `ResourceRenderer` , we mud the waters - the resource renderer now knows about vmis / cluster configurations / options / etc. IMO, that defeats the whole purpose of the split. 

@iholder-redhat your thoughts ? Can you spot an intermediate solution so we can somehow meet in the middle ?

Reply:
That makes sense.
I think we can keep it this way